src/auth/auth.service.ts

```diff
@@ -0,0 +1,40 @@
+import { JwtService } from '@nestjs/jwt';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+
+import { LoginDto } from './dto/login.dto';
+import { UsersService } from '../users/users.service';
+import { CreateUserDto } from '../users/dto/create-user.dto';
+import { JwtPayload } from '../common/interfaces/jwt-payload.interface';
+import * as bcrypt from 'bcrypt';
+import { User } from '../users/entities/user.entity';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UsersService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  signUp(createUserDto: CreateUserDto) {
+    return this.userService.create(createUserDto);
+  }
+
+  async logIn(loginDto: LoginDto): Promise<{ accessToken: string }> {
+    const { email, password } = loginDto;
+    const user = await this.userService.getFullUserData(email);
+    if (user && (await this.validatePassword(password, user))) {
+      const payload: JwtPayload = { email, fullName: user.fullName };
+      const accessToken: string = await this.jwtService.sign(payload);
+      return { accessToken };
+    }
+    throw new UnauthorizedException('Invalid credentials!');
+  }
+
+  private async validatePassword(
+    password: string,
+    user: User,
+  ): Promise<boolean> {
+    const hash = await bcrypt.hash(password, user.salt);
```

Comment:
https://www.npmjs.com/package/bcrypt#to-check-a-password

Reply:
@tar-aldev  can you explain? I am not quiet understood